art/defences/trainer/certified_trainer_pytorch.py

```diff
@@ -0,0 +1,372 @@
+        bound: float = 0.1,
+        loss_weighting: float = 0.1,
+        batch_size: int = 10,
+        concrete_to_zonotope: Optional[Any] = None,
+        use_certification_schedule: bool = True,
+        certification_schedule: Optional[Any] = None,
+        pgd_params: Optional["PGDParamDict"] = None,
+    ) -> None:
+        """
+        Create an :class:`.AdversarialTrainerCertified` instance.
+
+        Default values are for MNIST in pixel range 0-1.
+
+        :param classifier: Classifier to train adversarially.
+        :param pgd_params: A dictionary containing the specific parameters relating to regular PGD training.
+                           Must contain  the following keys:
+
+                               "eps": Maximum perturbation that the attacker can introduce.
+                               "eps_step": Attack step size (input variation) at each iteration.
+                               "max_iter": The maximum number of iterations.
+                               "batch_size": Size of the batch on which adversarial samples are generated.
+                               "num_random_init": Number of random initialisations within the epsilon ball.
+                                                  For num_random_init=0 starting at the original input.
+
+                           If not provided, we will default to typical MNIST values.
+        :param bound: The perturbation range for the zonotope. Will be ignored if a certification_schedule is used.
+        :param loss_weighting: Weighting factor for the certified loss.
+        :param concrete_to_zonotope:  Optional argument. Function which takes in a concrete data point and the bound
+                                      and converts the datapoint to the zonotope domain via:
+
+                                                processed_sample = concrete_to_zonotope(sample, bound)
+
+                                      If left as None, by default we apply the bound to every feature equally and
+                                      adjust the zonotope such that it remains in the 0 - 1 range.
+
+        :param nb_epochs: Number of training epochs.
+        :param use_certification_schedule: If to use a training schedule for the certification radius.
+        :param certification_schedule: Schedule for gradually increasing the certification radius. Empirical studies
+                                       have shown that this is often required to achieve best performance.
+                                       Either True to use the default linear scheduler,
+                                       or a class with a .step() method that returns the updated bound every epoch.
+
+        :param batch_size: Size of batches to use for certified training. NB, this will run the data
+                           sequentially accumulating gradients over the batch size.
+        """
+        super().__init__(classifier=classifier)  # type: ignore
```

Comment:
Why is this type ignore required?

Reply:
removed redundant ignore: was for an older version of the code